hack/kserve_migration/kserve_migration.sh

```diff
@@ -0,0 +1,236 @@
+log INFO "updating virtual services with new owner reference"
+for (( i=0; i<${isvc_count}; i++ ));
+do
+    owner_ref_count=$(kubectl get virtualservices ${isvc_names[$i]} -n ${isvc_ns[$i]} -o json | jq --raw-output '.metadata.ownerReferences | length')
+    if [ $owner_ref_count -eq 0 ]; then
+        isvc_uid=${infr_uid_map[${isvc_names[$i]}]}
+        kubectl patch virtualservices ${isvc_names[$i]} -n ${isvc_ns[$i]} --type='json' -p='[{"op": "add", "path": "/metadata/ownerReferences", "value": [{"apiVersion": "serving.kserve.io/v1beta1","blockOwnerDeletion": true,"controller": true,"kind": "InferenceService","name": "'${isvc_names[$i]}'","uid": "'${isvc_uid}'"}] }]'
+    fi
+done
+sleep 5
+
+# Verify that all inference services are migrated and ready
+log INFO "verifying inference services are migrated and ready"
+for (( i=0; i<${isvc_count}; i++ ));
+do
+    (
+        trap 'log ERROR "inference service ${isvc_names[$i]} did not migrate properly. migration job exits with code 1."' ERR
+        kubectl wait --for=condition=ready --timeout=10s inferenceservice.serving.kserve.io/${isvc_names[$i]} -n ${isvc_ns[$i]}
+    )
+done
+
+# Start kfserving controller for clean up
+log INFO "starting kfserving controller for clean up"
+kubectl scale --replicas=1 statefulset.apps kfserving-controller-manager -n kfserving-system
+
+# Delete inference services running on kfserving
+log INFO "deleting inference services on kfserving"
+for (( i=0; i<${isvc_count}; i++ ));
+do
+    kubectl delete inferenceservice.serving.kubeflow.org ${isvc_names[$i]} -n ${isvc_ns[$i]}
+done
+
+# Clean up kfserving
+if [ "${CLEAN_KFSERVING}" == "true" ]; then
+    log INFO "deleting kfserving namespace"
+    kubectl delete ns kfserving-system
+
+    log INFO "deleting kfserving cluster role and cluster role binding"
+    kubectl delete ClusterRoleBinding kfserving-manager-rolebinding
+    kubectl delete ClusterRoleBinding kfserving-models-web-app-binding
+    kubectl delete ClusterRoleBinding kfserving-proxy-rolebinding
+
+    log INFO "deleting kfserving webhook configuration and crd"
+    kubectl delete CustomResourceDefinition inferenceservices.serving.kubeflow.org
+    kubectl delete CustomResourceDefinition trainedmodels.serving.kubeflow.org
+
+    kubectl delete MutatingWebhookConfiguration inferenceservice.serving.kubeflow.org
+    kubectl delete ValidatingWebhookConfiguration inferenceservice.serving.kubeflow.org
+    kubectl delete ValidatingWebhookConfiguration trainedmodel.serving.kubeflow.org
+fi
+
+log INFO "deleting kfserving revisions"
+for (( i=0; i<${isvc_count}; i++ ));
+do
+    revisions=$(kubectl get deployment.apps -n ${isvc_ns[$i]} -o json | jq -r '.items[] | select(.status.availableReplicas==null and .metadata.ownerReferences[0].kind=="Revision") | .metadata.ownerReferences[0].name')
+    for revision in "${revisions[@]}"
+    do
+        kubectl delete revision.serving.knative.dev ${revision} -n ${isvc_ns[$i]}
```

Comment:
why do we need to delete the revisions?

Reply:
It's optional. The old revision are active even after deleting kfserving namespace. I added steps clean up revisions as well.

Reply:
knative revisions are managed by knative services so we should't need to worry about its lifecycle.

Reply:
Okay @yuzisun . I will remove this step.